### PR TITLE
feat(cli) Add Option --filterKeyword

### PR DESCRIPTION
**Pull request by codingluke**
This makes it possible to select certain pages by a keyword to be added to the pdf.

```markdown
---
keywords:
  - pdf
---

# A docusaurus page to be added to the pdf
```

```bash
npx docs-to-pdf --filterKeyword='pdf' --initialDocURLs="https://docusaurus.io/docs/" --contentSelector="article" --paginationSelector="a.pagination-nav__link.pagination-nav__link--next" --excludeSelectors=".margin-vert--xl a,[class^='tocCollapsible'],.breadcrumbs,.theme-edit-this-page" 
```

## Commitlint fix

I was not able to commit caused by the following commitlint error. 
- then I figured out it is because the pre-commit hook is triggered before git is creating the file "COMMIT_EDITMSG".
- So I changed it to a `commit-msg` hook, which is triggered **after** the file is created. Now it works also on my machine.

```
$ commitlint --from=HEAD~1 --edit ''
~/Projects/github/docs-to-pdf/node_modules/@commitlint/cli/lib/cli.js:123
        throw err;
        ^

[Error: ENOENT: no such file or directory, open '~/Projects/github/docs-to-pdf/.git/COMMIT_EDITMSG'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '~/Projects/github/docs-to-pdf/.git/COMMIT_EDITMSG'
}
```

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -49,8 +50,10 @@ export async function generatePDF({
   headerTemplate,
   footerTemplate,
   protocolTimeout,
+  filterKeyword,
 }: GeneratePDFOptions): Promise<void> {
-  const execPath = process.env.PUPPETEER_EXECUTABLE_PATH ?? puppeteer.executablePath('chrome');
+  const execPath =
+    process.env.PUPPETEER_EXECUTABLE_PATH ?? puppeteer.executablePath('chrome');
```
Comment (codingluke) [inline src/utils.ts:56]: prettier

File: src/utils.ts [hunk 2]
```diff
@@ -59,7 +62,10 @@ export async function generatePDF({
     protocolTimeout: protocolTimeout,
   });
 
-  const chromeTmpDataDir = browser.process()?.spawnargs.find((arg) => arg.startsWith('--user-data-dir'))?.split('=')[1] as string;
+  const chromeTmpDataDir = browser
+    .process()
+    ?.spawnargs.find((arg) => arg.startsWith('--user-data-dir'))
+    ?.split('=')[1] as string;
```
Comment (codingluke) [inline src/utils.ts:68]: prettier

File: src/utils.ts [hunk 3]
```diff
@@ -81,14 +87,18 @@ export async function generatePDF({
...
+      } else if (filterKeyword && !(await matchKeyword(page, filterKeyword))) {
+        console.log(
+          chalk.yellowBright(
+            `Page excluded by keyword filter: ${filterKeyword}`,
+          ),
+        );
       } else {
-        contentHTML += html;
+        // Get the HTML string of the content section.
+        contentHTML += await getHtmlContent(page, contentSelector);
```
Comment (codingluke) [inline src/utils.ts:101]: I moved the getHtmlContent here, as it only has to be done when the page is not excluded...

File: tests/utils.spec.ts [hunk 4]
```diff
@@ -254,3 +268,38 @@ describe('replaceHeader', () => {
...
+  });
+
+  it('should be true with a existing filterKeyword', async () => {
+    expect(await matchKeyword(page, 'match')).toBe(true);
+  });
+
+  it('should be false with a nonexisting filterKeyword', async () => {
+    expect(await matchKeyword(page, 'no-match')).toBe(false);
+  });
+});
```
Comment (codingluke) [inline tests/utils.spec.ts:305]: I only added this test, the rest is just prettier

File: package.json
Comment (jean-humann) [inline package.json:1]: @codingluke No need to change the version here, auto-release will take do it directly

File: src/utils.ts (hunk 3 above)
Comment (jean-humann) [reply src/utils.ts:101]: Good catch !

**Review by jean-humann — COMMENTED** [review]
@codingluke Thank you for your contribution.
Could you just revert the version change in `package.json` and I will approve it !

**Comment by jean-humann** [comment]
@codingluke please update your branch and rebase from master to fix the issue with checks CI not triggering on your PR.

File: package.json
Comment (codingluke) [reply package.json:1]: I wondered how to handle the version. I am very late in open-source contributing, lerning a lot 😅

**Comment by codingluke** [comment]
@jean-humann the problem is, that there is no test for generatePDF. This leads to a coverage decrease on every change. I did not have the feeling that my PRs should introduce a generatePDF test even though it might be someting to thing about.

**Comment by jean-humann** [comment]
@codingluke you're right, I will fix it with this [PR](https://github.com/jean-humann/docs-to-pdf/pull/171).

**Comment by jean-humann** [comment]
@codingluke, fix is merge, if you rebase all your PR this afternoon I will merge those tonight 👍

**Comment by codingluke** [comment]
@jean-humann I just did it through github, it created a merge commit, does it bother you?

**Pull request merged**